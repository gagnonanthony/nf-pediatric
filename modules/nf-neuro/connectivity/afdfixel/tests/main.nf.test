nextflow_process {

    name "Test Process CONNECTIVITY_AFDFIXEL"
    script "../main.nf"
    process "CONNECTIVITY_AFDFIXEL"

    tag "modules"
    tag "modules_nfcore"
    tag "connectivity"
    tag "connectivity/afdfixel"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "connectivity.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("connectivity - afdfixel - no params") {
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test' ], // meta map
                        file("\${test_data_directory}/decompose.h5"),
                        file("\${test_data_directory}/fodf.nii.gz")
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("connectivity - afdfixel - with params") {
        config "./nextflow.config"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test' ], // meta map
                        file("\${test_data_directory}/decompose.h5"),
                        file("\${test_data_directory}/fodf.nii.gz")
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("connectivity - afdfixel - with stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test' ], // meta map
                        file("\${test_data_directory}/decompose.h5"),
                        file("\${test_data_directory}/fodf.nii.gz")
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }
}
