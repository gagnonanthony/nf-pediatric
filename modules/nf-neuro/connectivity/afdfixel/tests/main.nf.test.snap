{
    "connectivity - afdfixel - no params": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test__afd_fixel.h5:md5,f0595d6b2271563976d7eb25ab6ace49"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d2ad834c51f57ac698ccfa530eb041a2"
                ],
                "hdf5": [
                    [
                        {
                            "id": "test"
                        },
                        "test__afd_fixel.h5:md5,f0595d6b2271563976d7eb25ab6ace49"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2ad834c51f57ac698ccfa530eb041a2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-03T14:24:25.471712"
    },
    "connectivity - afdfixel - with params": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test__afd_fixel.h5:md5,3e3fc635f787f12d31a201ed6fce3448"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d2ad834c51f57ac698ccfa530eb041a2"
                ],
                "hdf5": [
                    [
                        {
                            "id": "test"
                        },
                        "test__afd_fixel.h5:md5,3e3fc635f787f12d31a201ed6fce3448"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2ad834c51f57ac698ccfa530eb041a2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-03T14:25:16.561475"
    },
    "connectivity - afdfixel - with stub": {
        "content": [
            [
                "versions.yml:md5,d2ad834c51f57ac698ccfa530eb041a2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-29T23:33:55.212985"
    }
}
